lib/helper/WebDriverIO.js

```diff
@@ -380,8 +383,18 @@ class WebDriverIO extends Helper {
    * this.helpers['WebDriverIO']._locate({name: 'password'}).then //...
    * ```
    */
-  _locate(locator) {
-    return this.browser.elements(withStrictLocator(locator));
+  _locate(locator, smartWait = false) {
+
+    if (!this.options.smartWait || !smartWait) return this.browser.elements(withStrictLocator(locator));
+    let els;
+
+    return this.defineTimeout({implicit: this.options.smartWait})
+      .then(() => this.debugSection('SmartWait', `Locating ${locator} in ${this.options.wait}`))
+      .then(() => {
+        return els = this.browser.elements(withStrictLocator(locator));
```

Comment:
I agree to this point but implicit wait doesn't check for visibility. 
If an element is on a page but is not visible better to fail with error "Element is invisible".
Otherwise, user may think that SmartWait just doesn't work.